app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -36,6 +37,15 @@ function parseName(fullName) {
 
 export function buildResolver(baseName) {
   return Ember.DefaultResolver.extend({
+    buildCustomResolveTrie() {
+      this._customResolveTrie = new SuffixTrie("/");
```

Comment:
Did you check if this slows down tests? I think they build new resolvers frequently.

I normally would not advise this for memory leak purposes, but the `requirejs.entries` list should never change. What if we kept the trie in a module local variable rather than making a new one each time `buildResolver` is called?

Reply:
Overall, this PR improves the app boot time, so it should actually improve the speed of tests, not slow them down.

The run time of the whole suite varies too much to draw conclusions from, but I was able to measure a definite improvement by running a subset of the tests (The `Acceptance: Bookmarking` module). Before this PR, I was averaging `17.6s (± 0.1s)` for that module. After this change, I average `17.1s (± 0.1s)` for the module. (averages over 5 runs, `±` represents the range of results)

Using a module-local variable might improve speeds even more, but I think it will be negligible. Other [resolver-related caches](https://github.com/ember-cli/ember-resolver/blob/e6e436b0a73abc716a3596ba6ec0c6b098eb26c8/addon/resolvers/classic/index.js#L149-L155) are local to each resolver object, and not persisted at a global/module level. I'd prefer to stick to that, to avoid any potential state leaks in future.

Reply:
On second thoughts, I'm going to go with @eviltrout's suggestion. It will give us a slight speed boost, and it will also work around a memory leak issue. Right now our Ember CLI test suite is leaking resolvers. Each SuffixTrie is about 800kb, so adding one of those to every resolver instance causes a large increase in memory use during tests (and makes Firefox crash).

Of course, we should aim to fix that resolver leak in the test suite. But I don't want that work to hold up this PR.